### PR TITLE
[#262] Fix app analytics submit redirect

### LINKED ISSUE #262: Unable to retreive Analytics, Always showing page not found
pagenot found
After apigee edge connection setup
Below are working
1. Apps
2. Products
3. Catalogs

Not working is
1. Analytics. whenever i search for analytics its showing me page not found. Sometimes its showming me no errors other than page not found sometimes i got as error on [transaltion.[](url)](url)

### PR DESCRIPTION
**Pull request by arlina-espinoza**
Fixes #262 .

**Current bug**
When submitting an app analytics form (eg: _user/2/apps/test_app_1/analytics_), a redirect is issued that prepends the site's base path to the URL. If the site is installed under the root directory (like _example.com/_), then the redirect works, but if the site is under a subdirectory (like _/example.com/drupal/folder/_), then the redirect goes to a page not found _(example.com/drupal/folder/drupal/folder/user/2/apps/test_app_1/analytics)_.

**Expected behavior**
This PR fixes the redirect so it goes to the correct URL (ie, _example.com/drupal/folder/user/2/apps/test_app_1/analytics_).

### CODE REVIEW

**Review by cnovak — APPROVED** [review]
LGTM

File: src/Form/AppAnalyticsFormBase.php [hunk 1]
```diff
@@ -445,14 +445,16 @@ protected function generateResponse(array &$form, AppInterface $app, string $met
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
-    $url = $this->urlGenerator->generateFromRoute('<current>', [], [
+    $route = \Drupal::routeMatch();
```
Comment (mxr576) [inline src/Form/AppAnalyticsFormBase.php]: Pls dependency inject the route matcher service.

**Review by mxr576 — CHANGES_REQUESTED** [review]
Tbh, this fix looks odd to me because all Drupal core forms support subdirs but none of them using this `getRawParameters()` parameters trick which indicates to me that there should be a better way to resolve this problem. 
image

**Comment by cnovak** [comment]
I see `NodeForm` `preview()` following this pattern but only to add parameters, not to pass parameters through to the redirect. Maybe you do not need to set the parameters & options again when using `$form_state->setRedirect`

> Tbh, this fix looks odd to me because all Drupal core forms support subdirs but none of them using this `getRawParameters()` parameters trick which indicates to me that there should be a better way to resolve this problem.
> image

File: src/Form/AppAnalyticsFormBase.php (hunk 1 above)
Comment (cnovak) [reply src/Form/AppAnalyticsFormBase.php]: I wasn't sure if we need to inject services when using them for one off situations like this, but I do agree with @mxr576 after looking at [coding standards](https://www.drupal.org/docs/8/api/services-and-dependency-injection/services-and-dependency-injection-in-drupal-8#s-using-dependency-injection)

> Dependency injection is the preferred method for accessing and using services in Drupal 8 and should be used whenever possible.

**Review by cnovak — CHANGES_REQUESTED** [review]
1. Add service through dependency injection
2. Verify need to pass parameters into $form_state->setRedirect

**Comment by arlina-espinoza** [comment]
@cnovak Pushed an update to this PR. Figured out that setting the form redirect directly to` "<current>" `avoids having to pull out the route name and route parameters altogether, so there is no need to inject another service. Please review again.

>     1. Add service through dependency injection
> 
>     2. Verify need to pass parameters into $form_state->setRedirect

File: src/Form/AppAnalyticsFormBase.php (hunk 1 above)
Comment (arlina-espinoza) [reply src/Form/AppAnalyticsFormBase.php]: This should not be needed  - see https://github.com/apigee/apigee-edge-drupal/pull/266#issuecomment-534248353.

**Review by cnovak — APPROVED** [review]
Looks good

**Pull request merged**